xc/xc7/yosys/utils.tcl

```diff
@@ -0,0 +1,21 @@
+# Update the CLKOUT[0-5]_PHASE and CLKOUT[0-5]_DUTY_CYCLE parameter values.
+# Due to the fact that Yosys doesn't support floating parameter values
+# i.e. treats them as strings, the parameter values need to be multiplied by 1000
+# for the PLL registers to have correct values calculated during techmapping.
+proc update_param { param_name cell } {
+    set param_value [getparam $param_name $cell]
+    if {$param_value ne ""} {
+	set new_param_value [expr int(round($param_value * 1000))]
```

Comment:
There is one issue with that multiplier - its different for PHASE and DUTY_CYCLE parameters. For phase its 1000 but for DUTY_CYCLE its 100*1000=100000. I think that it is due to my mistake when adapting the Verilog code that calculates PLL register content. The code required duty cycle as percent (0 - 100), hence the additional x100. Vivado requires it to be given as a fraction (0.0 - 1.0).

The issue can be corrected here or in the PLL techmap.

Reply:
Ok, corrected the multiplier value.